R/pkg/R/generics.R

```diff
@@ -439,6 +439,10 @@ setGeneric("covar_samp", function(col1, col2) {standardGeneric("covar_samp") })
 #' @export
 setGeneric("covar_pop", function(col1, col2) {standardGeneric("covar_pop") })
 
+#' @rdname dapply
+#' @export
+setGeneric("dapply", function(x, func, schema = NULL) { standardGeneric("dapply") })
```

Comment:
@davies,
1) Yes, we should optimize multiple successive calls to dapply(). It is a following-up task which has already a JIRA for it: https://issues.apache.org/jira/browse/SPARK-12923. 
2) We can state that DataFrames from non-schema dapply() contain serialized R data, but the format is private, user should have no dependency on it. We can add rules in Catalyst that only dapply() and collect() are allowed on such DataFrames. Other operations including saving to files are not allowed.

Anyway, I am OK if we do not support non-schema dapply() for now. I am waiting for your vote:) @shivaram, @davies, @rxin